### ISSUE TITLE
Gradient Error in BatchNormLayer

### ISSUE THREAD

**Issue by SantaTitular** [josiahwsmith10/complextorch#3]
Hi again,

I was working with some 1D conv layers and I got the following error when using BatchNormLayers: Without them, it works fine but I get a very low accuracy in comparison with the real-valued equivalent... 

![image](https://github.com/josiahwsmith10/complextorch/assets/68435474/5db69373-2f51-47c7-b8ec-7c06a58bbe2b)

I'm having a really hard time trying to figure out the problem. It is just a simple 1D cvnn used for binary classification as below:

```
class CVMaxpool1D(nn.MaxPool1d):

    def __init__(self, kernel_size) -> None:
        super().__init__(kernel_size)

    def forward(self, input: CVTensor) -> CVTensor:
        return cvnn.functional.apply_complex_split(super().forward, super().forward, input)

class CVConv1D_alternative(nn.Module):
    def __init__(
        self,
        in_features,
        hidden_features=None,
        out_features=None,
        act_layer=cvnn.CReLU(),
        drop=0.0,
    ):
        super().__init__()
        
        self.conv1 = cvnn.CVConv1d(1, 32, 3)
        self.act = cvnn.CReLU()
        self.bn1 = cvnn.CVBatchNorm1d(32)
        self.pool1 = CVMaxpool1D(kernel_size=2)

        self.conv2 =cvnn.CVConv1d(in_channels=32, out_channels=64, kernel_size=3)
        # cvnn.CReLU(),
        self.bn2 =cvnn.CVBatchNorm1d(64)
        self.pool2 =CVMaxpool1D(kernel_size=2)

        self.conv3 =cvnn.CVConv1d(in_channels=64, out_channels=128, kernel_size=3)
        # cvnn.CReLU(),
        self.bn3 =cvnn.CVBatchNorm1d(128)
        self.pool3 =CVMaxpool1D(kernel_size=2)

        self.conv4 =cvnn.CVConv1d(in_channels=128, out_channels=256, kernel_size=3)
        # cvnn.CReLU(),
        self.bn4 =cvnn.CVBatchNorm1d(256)
        self.pool4 =CVMaxpool1D(kernel_size=2)

        self.drop = cvnn.CVDropout(dropout)#  Dro.Dropout(dropout),
        
        self.fc1 = cvnn.CVLinear(25088, 1000)
        # cvnn.CVDropout(dropout),
        # cvnn.CReLU(),
        self.fc2 = cvnn.CVLinear(1000, out_features)

    def forward(self, x):
        x = x.reshape(x.shape[0],1,x.shape[1])#x = x.unsqueeze(1)  # Add channel dimension
        x = CVTensor(x.real, x.imag)  # batch_size x N
        # x = self.ConvNet(x)
        
        x = self.conv1(x)
        x = self.act(x)
        x = self.bn1(x)
        x = self.pool1(x)
        
        x = self.conv2(x)
        x = self.act(x)
        x = self.bn2(x)
        x = self.pool2(x)
        
        x = self.conv3(x)
        x = self.act(x)
        x = self.bn3(x)
        x = self.pool3(x)
        
        x = self.conv4(x)
        x = self.act(x)
        x = self.bn4(x)
        x = self.pool4(x)
        x = self.drop(x)
        
        x = x.view(-1,256*98)
        
        # x = self.task(x)
        
        x = self.fc1(x)
        x = self.act(x)
        x = self.drop(x)
        x = self.fc2(x)
        x = x.abs()
        
        return x

class complexTorchDataset(torch.utils.data.Dataset):

    def __init__(self, data):
        """
        data: the dict returned by utils.load_classification_data
        """
        train_X, train_y = data["train"]
        dev_X, dev_y = data["dev"]
        test_X, test_y = data["test"]
        
        self.X = torch.tensor(train_X, dtype=torch.complex64)
        self.y = torch.tensor(train_y, dtype=torch.long)

        self.dev_X = torch.tensor(dev_X, dtype=torch.complex64)
        self.dev_y = torch.tensor(dev_y, dtype=torch.long)

        self.test_X = torch.tensor(test_X, dtype=torch.complex64)
        self.test_y = torch.tensor(test_y, dtype=torch.long)

    def __len__(self):
        return len(self.X)

    def __getitem__(self, idx):
        return self.X[idx], self.y[idx]

def train_batch(X, y, model, optimizer, criterion, scheduler=False, **kwargs):

    model.train()
    optimizer.zero_grad()
    out = model(X, **kwargs)
    loss = criterion(out, y)
    loss.backward()
    optimizer.step()
    if scheduler:
        scheduler.step()
    return loss.item()

@torch.no_grad()
def evaluate(model, X, y, criterion):

    model.eval()
    logits = model(X)
    loss = criterion(logits, y)
    loss = loss.item()
    y_hat = logits.argmax(dim=-1)#
    n_correct = (y == y_hat).sum().item()
    n_possible = float(y.shape[0])
    model.train()
    return loss, n_correct / n_possible

data = {"train": (X_train, y_train),
                "dev": (X_dev, y_dev),
                "test": (X_test, y_test)}
        
        dataset = complexTorchDataset(data)
        train_dataloader = DataLoader(
            dataset, batch_size=batch_size, shuffle=True)
   
        
        dev_X, dev_y = dataset.dev_X, dataset.dev_y
        test_X, test_y = dataset.test_X, dataset.test_y
        
        n_classes = 2
        n_feats = dataset.X.shape[1]

        CVMLP_model = CVConv1D_alternative(n_feats,
                out_features=n_classes,
                hidden_features=hidden_size[0],
                drop=dropout,
                act_layer=cvnn.CReLU
            )

        optimizer = optim_cls(
            CVMLP_model.parameters(), lr=learning_rate, weight_decay=l2_decay
        )
        criterion = nn.CrossEntropyLoss()# complextorch.nn.modules.loss.SplitL1()# nn.BCELoss()

        # training loop
        epochs = torch.arange(1, epochs + 1)
        train_mean_losses = []
        valid_accs = []
        train_losses = []
        val_losses = []
        for ii in epochs:
            print('Training epoch {}'.format(ii))
            for X_batch, y_batch in train_dataloader:
                loss = train_batch(
                    X_batch, y_batch, CVMLP_model, optimizer, criterion, scheduler=scheduler)
                train_losses.append(loss)

            mean_loss = torch.tensor(train_losses).mean().item()
            print('Training loss: %.4f' % (mean_loss))

            train_mean_losses.append(mean_loss)
            val_loss, val_acc = evaluate(CVMLP_model, dev_X, dev_y , criterion)
            valid_accs.append(val_acc)
            val_losses.append(val_loss)
            print('Valid acc: %.4f' % (valid_accs[-1]))
        
        _, test_acc = evaluate(CVMLP_model, test_X, test_y, criterion)
        print('Final Test acc: %.4f' % (test_acc))
```

Thanks in advance!
Cheers
